tests/unit/endpoint_test.py

```
@@ -0,0 +1,163 @@
+    
+    # Verify port mapping
+    assert endpoint.get_port() == expected_port
+    
+    # Verify all fields are mapped correctly (not just port)
+    assert endpoint.get_address() == b"127.0.1.1", "Address must be mapped from proto"
+    assert endpoint.get_domain_name() == "redpanda.com", "Domain name must be mapped from proto"
+    
+    # Protect against breaking changes - PRIORITY 1
+    assert isinstance(endpoint, Endpoint), "Must return Endpoint instance"
+
+@pytest.mark.parametrize("field_to_none, attr_name, expected_default", [
```

Ruff (0.14.10)

Wrong type passed to first argument of `pytest.mark.parametrize`; expected `tuple`

Use a `tuple` for the first argument

(PT006)